### PR TITLE
add stream segment serialization.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @superyyrrzz @qinezh 
#449

### CODE REVIEW

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Utility/StreamSegmentSerialization/StreamSerializer.cs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            }
+            var dict = value as IReadOnlyDictionary<string, object>;
+            if (dict != null)
+            {
+                return WriteDictionary(dict);
+            }
+            throw new NotSupportedException();
+        }
+
+        private StreamSegment WriteString(string value)
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Utility/StreamSegmentSerialization/StreamSerializer.cs]: >WriteString [](start = 30, length = 11)

seems same to WriteBytes except the streamSegmentType?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Utility/StreamSegmentSerialization/StreamSerializer.cs]: Yes, but call different write overload.

---
In reply to: [78334947](https://github.com/dotnet/docfx/pull/687#discussion_r78334947) [](ancestors = 78334947)

**Comment by ansyral** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**